client.go

```diff
@@ -1875,6 +1875,10 @@ func (c *Client) RecursivePushFile(container string, source string, target strin
 	}
 
 	sourceDir := filepath.Dir(source)
+	sourceLen := len(sourceDir)
+	if sourceDir == "." {
+		sourceLen--
+	}
 
```

Comment:
Because `filepath.Split` doesn't return `.`. I would save some time to see tests on why it was added. I tested that everything works on my setup, but I am not aware of the use case when it was added.